lib/server.js

```diff
@@ -68,64 +67,111 @@ module.exports = Server;
  * @param {string} [opts.username] Username for RabbitMQ.
  * @param {string} [opts.password] Username for Password.
  */
-function Server(opts) {
+function Server (opts) {
+  this._tasks = {};
   this.opts = opts || {};
-  defaults(this.opts, Server.DEFAULT_OPTIONS);
+  defaults(this.opts, {
+    hostname: process.env.RABBITMQ_HOSTNAME || 'localhost',
+    port: process.env.RABBITMQ_PORT || 5672,
+    username: process.env.RABBITMQ_USERNAME || 'guest',
+    password: process.env.RABBITMQ_PASSWORD || 'guest'
+  });
 
-  if (!Array.isArray(this.opts.queues)) {
-    throw new Error('ponos.Server: missing required `queues` option.');
-  }
+  if (this.opts.hermes) {
+    this.hermes = this.opts.hermes;
+  } else {
+    if (!Array.isArray(this.opts.queues)) {
+      throw new Error('ponos.Server: missing required `queues` option.');
```

Comment:
whats your thoughts on using error cat to create the thrown error? so we get this in rollbar.
or do you think we will never hit the thrown case in prod since test *should* catch this?